modules/zstd/option.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package zstd
+
+import "github.com/klauspost/compress/zstd"
+
+type WriterOption = zstd.EOption
+
+var (
+	WithEncoderCRC               = zstd.WithEncoderCRC
+	WithEncoderConcurrency       = zstd.WithEncoderConcurrency
+	WithWindowSize               = zstd.WithWindowSize
+	WithEncoderPadding           = zstd.WithEncoderPadding
+	WithEncoderLevel             = zstd.WithEncoderLevel
+	WithZeroFrames               = zstd.WithZeroFrames
+	WithAllLitEntropyCompression = zstd.WithAllLitEntropyCompression
+	WithNoEntropyCompression     = zstd.WithNoEntropyCompression
+	WithSingleSegment            = zstd.WithSingleSegment
+	WithLowerEncoderMem          = zstd.WithLowerEncoderMem
+	WithEncoderDict              = zstd.WithEncoderDict
+	WithEncoderDictRaw           = zstd.WithEncoderDictRaw
+)
+
+type EncoderLevel = zstd.EncoderLevel
+
+const (
+	SpeedFastest           EncoderLevel = zstd.SpeedFastest
+	SpeedDefault           EncoderLevel = zstd.SpeedDefault
+	SpeedBetterCompression EncoderLevel = zstd.SpeedBetterCompression
+	SpeedBestCompression   EncoderLevel = zstd.SpeedBestCompression
+)
+
+type ReaderOption = zstd.DOption
+
+var (
+	WithDecoderLowmem      = zstd.WithDecoderLowmem
+	WithDecoderConcurrency = zstd.WithDecoderConcurrency
+	WithDecoderMaxMemory   = zstd.WithDecoderMaxMemory
+	WithDecoderDicts       = zstd.WithDecoderDicts
+	WithDecoderDictRaw     = zstd.WithDecoderDictRaw
+	WithDecoderMaxWindow   = zstd.WithDecoderMaxWindow
+	WithDecodeAllCapLimit  = zstd.WithDecodeAllCapLimit
+	WithDecodeBuffersBelow = zstd.WithDecodeBuffersBelow
+	IgnoreChecksum         = zstd.IgnoreChecksum
+)
```

Comment:
Do we really need all these aliases? I think it's better to use the upstream vars directly or else this needs to be kept in sync with upstream, which is useless extra work.

Reply:
It are just little tricks.

```go
import (
	"code.gitea.io/gitea/modules/zstd"
	compress_zstd "github.com/klauspost/compress/zstd"
)

func main() {
	writer, err := zstd.NewSeekableWriter(io.Discard, 10240, compress_zstd.WithEncoderLevel(compress_zstd.SpeedFastest))
	// ...
}
```

vs

```go
import (
	"code.gitea.io/gitea/modules/zstd"
)

func main() {
	writer, err := zstd.NewSeekableWriter(io.Discard, 10240, zstd.WithEncoderLevel(zstd.SpeedFastest))
	// ...
}
```

Reply:
I think I prefer the former for the sake of maintainability. The less first-party code, the better.

Reply:
Since the new `modules/zstd` is designed to wrap `klauspost/compress/zstd` and `zstd-seekable-format-go`, its work is to hide the underlying packages. And I don't see its cause more maintenance works. Maybe you are concerned about:

1. What if the name or header has been changed for those functions?
  It's almost not allowed for Golang packages with SemVer. If it happened, it's the upstream's fault to increase maintenance work, not the `modules/zstd`.
2. What if a function has been removed?
  It's the same, a breaking change. For Golang packages with SemVer, it must increase to v2 which can be treated as a new package.
3. What if a new function has been added?
  The new function must be obscure to be added so late. And if Gitea really need it, it's compatible to use it directly. Like
```go
import (
	"code.gitea.io/gitea/modules/zstd"
	compress_zstd "github.com/klauspost/compress/zstd"
)

func main() {
	writer, err := zstd.NewSeekableWriter(io.Discard, 10240,
		zstd.WithEncoderLevel(zstd.SpeedFastest),
		compress_zstd.WithNewOption(),
	)
	// ...
}
```
But believe me, most other maintainers don't like the two imports, and they will be happy to add `WithNewOption` to `modules/zstd` since the usage code is clearer, but won't complain that it's unnecessary maintenance work.

Reply:
BTW, I do agree "the less first-party code, the better.", but don't forget `compress_zstd "github.com/klauspost/compress/zstd"` is also a line of first-party code.

Or we could just give up `modules/zstd`, and yes, less first-party code in `modules`, but more first-party code in other places.

Reply:
Yeah I guess my concerns will mostly be covered by the go compiler, e.g. it would break the build if upstream breaks it by changing those symbols. Still, I find such copying of symbols mostly useless, but I can live with it.